keps/sig-auth/2579-psp-replacement/README.md

```diff
@@ -0,0 +1,1073 @@
+
+Notes:
+- This set is equal to the [docker default capability
+  set](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities) minus
+  `NET_RAW`.
+- The [OpenShift default capability
+  set](https://github.com/openshift/hypershift-toolkit/blob/148be6f31a365dbcdf1cbd647773f59ccbcc282a/assets/ignition/files/etc/crio/crio.conf#L87-L102)
+  also drops AUDIT_WRITE, MKNOD, SETFCAP, and SYS_CHROOT._
+- The allowed set for the restricted profile is implicitly empty (no capabilities), since
+  [Kubernetes does not support ambient
+  capabilities](https://github.com/kubernetes/kubernetes/issues/56374). If ambient capability
+  support is ever added, we may want to consider a more conservative allowed set for the restricted
+  profile.
+
+**Volumes** - Inline CSI volumes will be allowed by the restricted profile. Justification:
+  - Inline CSI volumes should only be used for ephemeral volumes
+  - The CSIDriver object spec controls whether a driver can be used inline, and can be modified
+    without binary changes to disable inline usage.
+  - Risky inline drivers should already use a 3rd party admission controller, since they are usable
+    by the baseline policy.
+  - We should thoroughly document safe usage, both on the documentation for this (pod security
+    admission) feature, as well as in the CSI driver documentation.
+
+**Windows Host Process** - [Privileged Windows container
+support](https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1981-windows-privileged-container-support)
+is targeting alpha in v1.22, and adds new fields for running privileged windows containers. These
+fields will be restricted under the baseline policy.
+- **Restricted Fields:**
+  - `spec.securityContext.windowsOptions.hostProcess`
+  - `spec.containers[*].securityContext.windowsOptions.hostProcess`
+- **Allowed Values:** false, undefined/nil
+
+_Note: These fields should be unconditionally restricted, regardless of targeted OS._
+
+### Windows Support
+
+In the initial alpha implementation, Windows pods will be supported by both the `privileged` and
+`baseline` profiles. Windows pods _may_ be broken by the restricted field, which requires setting
+linux-specific settings (such as seccomp profile, run as non root, and disallow privilege
+escalation). If the Kubelet and/or container runtime choose to ignore these linux-specific values at
+runtime, then windows pods should still be allowed under the restricted profile, although the
+profile will not add additional enforcement over baseline (for Windows).
+
+Windows support will be reevaluated prior to this policy feature going to beta, or if/when
+Kubernetes adds support to definitively distinguish between Windows and Linux workloads. See
+[Windows restricted profile support](#windows-restricted-profile-support) for more details.
+
+### Flexible Extension Support
+
+In order to make it as easy as possible to extend and customize this policy controller, we will
```

Comment:
I'd like to explicitly indicate that we consider the namespace labels a stable API and recommend other admission plugins use it to enforce policy.  I like CSI drivers as a good example of this.

Even if we aren't willing to state explicit examples, I'd like to confirm the usability of our namespace labels as stable APis for 3rd party admission plugins to depend upon.